test/e2e/cleanup/submariner.go

```diff
@@ -0,0 +1,264 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleanup
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sync"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/admiral/pkg/watcher"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	operatorv1alpha1 "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/controllers/uninstall"
+	operatorclient "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	operatorv1alpha1client "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned/typed/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/names"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	submarinerclientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+var _ = Describe("Submariner cleanup", func() {
+	When("the Submariner resource is deleted", testSubmarinerCleanup)
+})
+
+func testSubmarinerCleanup() {
+	_ = framework.NewFramework("submariner-cleanup")
+
+	var (
+		submarinerInterface  operatorv1alpha1client.SubmarinerInterface
+		submariner           *operatorv1alpha1.Submariner
+		routeAgentPodMonitor *podMonitor
+		gatewayPodMonitor    *podMonitor
+		brokerRestConfig     *rest.Config
+		stopCh               chan struct{}
+	)
+
+	BeforeEach(func() {
+		stopCh = make(chan struct{})
+
+		operatorClient, err := operatorclient.NewForConfig(framework.RestConfigs[framework.ClusterA])
+		Expect(err).To(Succeed())
+
+		submarinerInterface = operatorClient.SubmarinerV1alpha1().Submariners(framework.TestContext.SubmarinerNamespace)
+
+		submariner, err = submarinerInterface.Get(context.TODO(), submarinercr.SubmarinerName, metav1.GetOptions{})
+		Expect(err).To(Succeed())
+
+		brokerRestConfig = getBrokerRestConfig(submariner.Spec.BrokerK8sRemoteNamespace)
+		Expect(brokerRestConfig).ToNot(BeNil(), "No broker located")
+
+		routeAgentPodMonitor = startPodMonitor(names.AppendUninstall(names.RouteAgentComponent), stopCh)
```

Comment:
Shall we include globalnetPod as well, as it will cover even routeAgent and gatewayPods?

Reply:
yeah I was going to add globalnet as well.